src/guc.c

```diff
@@ -373,6 +374,18 @@ _guc_init(void)
 							 NULL,
 							 NULL);
 
+	DefineCustomBoolVariable("timescaledb.enable_compression_index_lookup",
+							 "Enable index lookup during compression",
+							 "Enable index lookup during compression, use the matching index"
+							 "to get the sorted tuples in order to save the sorting cpu cycles",
```

Comment:
+1. I don't think we need a guc to enable/disable this.

Reply:
I discuss the rationale of having toggle for this feature. I will help in providing workaround in case of regression in production. We can take a call in future once this gets stabilized.